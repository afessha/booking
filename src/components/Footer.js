import React from "react";
import "./Booking.css";
const Footer = () => {
  return <div className="footer"> Awet Fessha @2020 </div>;
};
export default Footer;
