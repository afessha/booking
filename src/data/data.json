[
  {
    "firstname": "Mozafar",
    "lastname": "Haider",
    "diningDate": "10-09-2020",
    "numberOfCovers": 5,
    "phoneNumber": "07456789542",
    "email": "mozafar@gmail.com"
  },
  {
    "firstname": "Awet",
    "lastname": "Fessha",
    "diningDate": "18-06-2020",
    "numberOfCovers": 3,
    "phoneNumber": "07454789145",
    "email": "awet@gmail.com"
  }
]
