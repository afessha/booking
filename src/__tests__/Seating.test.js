import React from "react";
import { render } from "@testing-library/react";
import { create } from "react-test-renderer";
import data from "../data/data.json";
import { Link, BrowserRouter } from "react-router-dom";
import Footer from "../components/footer/Footer";
class Seating extends React.Component {
  constructor() {
    super();
    this.state = {
      bookings: data,
      // filteredBookings: data,
      // searchValue: "",
    };
  }

  //   onChange = (e) => {
  //     const state = this.state;
  //     state[e.target.name] = e.target.value;
  //     this.setState(state);
  //     const { searchValue } = this.state;
  //     const filteredOutput = this.state.bookings.filter((result) => {
  //       return (
  //         result.firstname === searchValue || result.lastname === searchValue
  //       );
  //     });
  //     this.setState({ filteredBookings: filteredOutput });
  //   };

  render() {
    let users = JSON.parse(localStorage.getItem("users") || "[]");
    return (
      <BrowserRouter>
        <div>
          <h3 className="seating-title">Booking List </h3>
          {/* <input
          classname="search-input"
          placeholder="search by name"
          onChange={this.onChange}
        /> */}
          <div className="link-to-booking">
            <Link to={"/"}>Go to Booking page</Link>
          </div>
          <br />
          <br />

          <table className="table table-hover">
            <thead>
              <tr>
                <th scope="col">First Name </th>
                <th scope="col">Last Name </th>
                <th scope="col">Dining Date</th>
                <th scope="col">Number of Covers </th>
                <th scope="col">Phone Number</th>
                <th scope="col">Email</th>
                <th scope="col">Status</th>
              </tr>
            </thead>
            {users.map((result) => (
              <tbody>
                <tr>
                  <td>{result.firstname} </td>

                  <td>{result.lastname} </td>

                  <td>{result.diningDate} </td>

                  <td>{result.numberOfCovers} </td>

                  <td>{result.phoneNumber}</td>

                  <td>{result.email}</td>
                  <td>
                    {" "}
                    <select>
                      <option value="notArrived">Not arrived</option>
                      <option value="seated">Seated</option>
                    </select>
                  </td>
                </tr>
              </tbody>
            ))}
          </table>
          <Footer />
        </div>
      </BrowserRouter>
    );
  }
}

export default Seating;

describe("Seating component", () => {
  test("Matches the snapshot", () => {
    const seating = create(<Seating />);
    expect(seating.toJSON()).toMatchSnapshot();
  });
});
