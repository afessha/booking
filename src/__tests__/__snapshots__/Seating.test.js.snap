// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Matches the snapshot 1`] = `
<div>
  <h3
    className="seating-title"
  >
    Booking List 
  </h3>
  <div
    className="link-to-booking"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Go to Booking page
    </a>
  </div>
  <br />
  <br />
  <table
    className="table table-hover"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          First Name 
        </th>
        <th
          scope="col"
        >
          Last Name 
        </th>
        <th
          scope="col"
        >
          Dining Date
        </th>
        <th
          scope="col"
        >
          Number of Covers 
        </th>
        <th
          scope="col"
        >
          Phone Number
        </th>
        <th
          scope="col"
        >
          Email
        </th>
        <th
          scope="col"
        >
          Status
        </th>
      </tr>
    </thead>
  </table>
  <div
    className="footer"
  >
     Awet Fessha @2020 
  </div>
</div>
`;

exports[`Seating component Matches the snapshot 1`] = `
<div>
  <h3
    className="seating-title"
  >
    Booking List 
  </h3>
  <div
    className="link-to-booking"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Go to Booking page
    </a>
  </div>
  <br />
  <br />
  <table
    className="table table-hover"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          First Name 
        </th>
        <th
          scope="col"
        >
          Last Name 
        </th>
        <th
          scope="col"
        >
          Dining Date
        </th>
        <th
          scope="col"
        >
          Number of Covers 
        </th>
        <th
          scope="col"
        >
          Phone Number
        </th>
        <th
          scope="col"
        >
          Email
        </th>
        <th
          scope="col"
        >
          Status
        </th>
      </tr>
    </thead>
  </table>
  <div
    className="footer"
  >
     Awet Fessha @2020 
  </div>
</div>
`;
